helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterContext.java

```diff
@@ -104,6 +106,15 @@
     _estimatedMaxTopStateCount = estimateAvgReplicaCount(totalTopStateReplicas, instanceCount);
     _baselineAssignment = baselineAssignment;
     _bestPossibleAssignment = bestPossibleAssignment;
+
+    _busyInstances = _bestPossibleAssignment.values().stream().flatMap(
+        resourceAssignment -> resourceAssignment.getRecord().getMapFields().values().stream()
```

Comment:
This is a bit of general advice FYI. 
I feel concerned about manipulating a map of nested map data structure in multiple places, it's in general hard to read & follow. A centralized data model of bestPossibleAssignement, baselineAssignment, even current assignment would be ideal. Something like
```
public class ClusterAssignment {
   
   public xxx findDifference(ClusterAssignment other) {
   }
   public Set<String> getBusyInstances() {
    .... 
   }
}
ClusterAssignment bestPossible;
ClusterAssignment baseLine;
```

It would be easier to unit test and have as many as things centralized in one place

Reply:
I agree. Just this seems to be a too large change for this PR. Let me try the best to make it clearer with less code change. I have moved the calculation to the algorithm implementation for now. Later, as you mentioned if we have more requirements, we can split it out to a class.